com.unity.sg2/Editor/GraphUI/EditorCommon/CommandStateObserver/ShaderGraphCommandOverrides.cs

```diff
@@ -35,18 +35,16 @@ public static class ShaderGraphCommandOverrides
 
         // TODO: (Sai) Move this stuff into ShaderGraphModel as much as possible
         // there are things that GTF is doing in the base command handler that we shouldn't be disabling or affecting at all
-        public static void HandleDeleteElements(
+        public static void HandleDeleteNodesAndEdges(
             UndoStateComponent undoState,
             GraphModelStateComponent graphModelState,
             SelectionStateComponent selectionState,
             PreviewManager previewManager,
             DeleteElementsCommand command)
         {
-            // Don't want to call base command handler as doing so wipes command from info. of objects being deleted
-            //DeleteElementsCommand.DefaultCommandHandler(undoState, graphViewState, selectionState, command);
-
-            if (!command.Models.Any())
-                return;
+            var modelsToDelete = command.Models.ToList();
+            // We want to override base handling here
+            // DeleteElementsCommand.DefaultCommandHandler(undoState, graphModelState, selectionState, command);
```

Comment:
Is this true? Don't we still need to delete the GTF models?

Reply:
We do still delete the models, we're just handling it a little differently due to our need to process redirect nodes first